File: completed/1957.java
import java.util.*;

class Solution {
    public String makeFancyString(String s) {
        StringBuilder sb = new StringBuilder();
        sb.append(s.charAt(0));

        int consecutive = 1;
        for (int i = 1; i < s.length(); i++) {
            if (s.charAt(i) == sb.charAt(sb.length() - 1)) {
                if (consecutive == 2) {
                    continue;
                }

                sb.append(s.charAt(i));
                consecutive++;
            } else {
                sb.append(s.charAt(i));
                consecutive = 1;
            }
        }
        return sb.toString();
    }
}

File: completed/1957.cpp
class Solution {
public:
    string makeFancyString(string s) {
        string fancy = "";
        for (int i = 0; i < s.length(); i++) {
            fancy += s[i];
            if (i + 1 >= s.length() || s[i + 1] != s[i]) {
                continue;
            }

            fancy += s[i];
            while (i + 1 < s.length() && s[i + 1] == s[i]) {
                i++;
            }
        }
        return fancy;
    }
};
